### ISSUE TITLE
Plans to support 3dof devices?

### ISSUE THREAD

**Issue by xdave** [zwin-project/zwin#33]
Not full vr headsets, but AR glasses like the XReal Air.

I know a few drivers exist for Linux for the Air:
- userspace hid driver: https://gitlab.com/TheJackiMonster/nrealAirLinuxDriver
  - recent fork: https://github.com/wheaney/xrealAirLinuxDriver
- openvr driver providing an hmd device: https://github.com/wheaney/OpenVR-xrealAirGlassesHMD
- xr driver inside of monado (merged mainline): https://gitlab.freedesktop.org/monado/monado/-/tree/main/src/xrt/drivers/nreal_air?ref_type=heads

Next week, in getting a pair of these, and I'm assessing my options for productivity purposes for these on Linux.

Thanks.

**Comment by Aki-7** [comment]
Thank you for the suggestion!
I want to expand the hardware compatibility for sure, but 3DoF devices are not so highly prioritized for now.

@rtr-dnd @shierote Do you have any opinion on this?

**Comment by rtr-dnd** [comment]
We should definitely consider supporting AR glasses!
However my concern with 3DoF devices is that texts and UI might become harder to read, as you can't get closer with 3DoF. 
So my opinion is it needs a better UI scaling method before supporting such devices. 
But optimistically, it might be fine if the device is placed correctly in the virtual world and the resolution of the device is high enough...?

**Comment by xdave** [comment]
Yeah. The XReal Air glasses have pretty good clarity, even at 1080p, but yeah, the UI scaling would need to be good.  There's proprietary software (called "Nebula") that works with it that provides virtual screens from the device into which it's plugged; but, support for it is limited.

This guy (https://github.com/wheaney/breezy-desktop) is developing support for virtual spatial screens as we speak, so you might be able to get some pointers from it and from the projects I posted above.
